components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/Constants.java

```diff
@@ -57,7 +57,10 @@ public enum ErrorMessage {
                 "The provided userId is invalid."),
         ERROR_CODE_SERVER_ERROR(USER_MANAGEMENT_PREFIX.getPrefix() + "15001",
                 "Unable to retrieve User.",
-                "Server Encountered an error while retrieving the user.");
+                "Server Encountered an error while retrieving the user."),
+        INVALID_TENANT_DOMAIN(USER_MANAGEMENT_PREFIX.getPrefix() + "10002",
```

Comment:
Seems like this is a server error. Shall we start the error code with `150xx` instead of `100xx`? We use `100xx` for client errors.

Reply:
INVALID_TENANT_DOMAIN is a client error IMO... as the client could change the context tenantDomain to a non-existing one.